src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,122 @@
+# we consider the ODE dxₜ/dt = f(xₜ,p(rt))
+# here p = p(t) ∈ Rᵐ is a function containing all the system parameters 
+
+# We ask the user to define: 
+#  1) a ContinuousTimeDynamicalSystem that should be investigated and
+#  2) a protocol for the time-dependent forcing with the struct RateConfig
+
+# Then we give back the ContinuousTimeDynamicalSystem with the parameter 
+# changing according to the rate protocol
+"""
+    RateConfig
+
+Time-dependent forcing protocol containing the information to apply a parameter shift to an autonomous system.
+
+Fields
+==============
+
+- p: monotonic function which describes the time-dependent parametric forcing 
+- p_parameters: the vector of parameters which are associated with p
+- t_pstart: the parameter values of the past limit system are given by p(p_parameteters,t_pstart)
+- t_pend: the parameter values of the future limit system are given by p(p_parameters,t_pend) 
+- t_start: the explicit value of time at which the nonautonomous dynamics starts 
+- t_ramp_length: the duration of time of nonautonomous dynamics [i.e. within (t_start,t_start+t_ramp_length)] 
+- dp: the difference in parameter values attained across the ramping
+
+Default values 
+==============
+
+- t_pstart = -100
+- t_pend = 100
+- p_parameters = []
+- t_start = -t_ramp_length/2
+- dp = 1
```

Comment:
I don't think these defaults make sense. Unless there is a singular unambiguous value, like 0 for starting time, you should not have defaults and request the user to provide them. 
`p_parameters` should be `nothing` by default as with the rest of DynamicalSYstems.jl